src/print.c

```diff
@@ -189,6 +189,46 @@ can_preview (void)
   return FALSE;
 }
 
+static GtkPrintCapabilities
+print_capabilities_from_options (GVariant *options)
+{
+  g_auto(GStrv) supported_output_file_formats = NULL;
+  GtkPrintCapabilities capabilities;
+
+  capabilities = can_preview () ? GTK_PRINT_CAPABILITY_PREVIEW : 0 |
+                 GTK_PRINT_CAPABILITY_PAGE_SET |
+                 GTK_PRINT_CAPABILITY_COPIES |
+                 GTK_PRINT_CAPABILITY_COLLATE |
+                 GTK_PRINT_CAPABILITY_REVERSE |
+                 GTK_PRINT_CAPABILITY_SCALE |
+                 GTK_PRINT_CAPABILITY_NUMBER_UP;
+
+  if (g_variant_lookup (options, "supported_output_file_formats", "^as", &supported_output_file_formats))
+    {
+      g_autoptr(GHashTable) formats = NULL;
+
+      g_assert (supported_output_file_formats && supported_output_file_formats[0] != NULL);
+
+      formats = g_hash_table_new (g_str_hash, g_str_equal);
+
+      for (size_t i = 0; supported_output_file_formats[i]; i++)
+        g_hash_table_add (formats, (gpointer) supported_output_file_formats[i]);
+
+      /* This is a limitation of GtkPrintCapability - it doesn't have an enum
+       * for SVG files.
+       */
+      if (!g_hash_table_contains (formats, "svg"))
```

Comment:
Is there a particular reason why you need to put the formats inside a set? We shouldn't be getting duplicates, and if you want to look for a string inside a string vector, you can use [`g_strv_contains()`](https://docs.gtk.org/glib/func.strv_contains.html).

Reply:
It's going to be a very small vector, so computationally isn't going to be a huge deal; we probably lose more time creating, populating, and destroying the hash table.

I don't mind using a hash table, just wanted to know if we were supposed to do some validation on the list of supported formats or not.